lib/Conversion/VerifToSMT/VerifToSMT.cpp

```diff
@@ -274,6 +274,16 @@ struct VerifBoundedModelCheckingOpConversion
       }
     }
 
+    // TODO: this can be removed once we have a way to associate reg ins/outs
+    // with clocks
+    if (clockIndexes.size() > 1)
+      return op->emitError(
+          "only modules with one clock are currently supported");
+
+    // If we have no clocks then num_regs has to be 0
+    if (clockIndexes.size() == 0 && numRegs != 0)
+      return op->emitError("num_regs is non-zero but found no clock");
```

Comment:
We shouldn't use `emitError` in conversion patterns. In theory, only if this pattern application fails doesn't mean that the whole pass fails. If. you. want error messages you need to do the checks in a walk before the conversion patterns are applied.

Reply:
Ah cool, makes sense! I'll swap to it to a pre-application check.